examples/cryptocurrency-advanced/backend/src/lib.rs

```diff
@@ -47,11 +51,16 @@ pub const INITIAL_BALANCE: u64 = 100;
 pub struct CryptocurrencyService;
 
 impl Service for CryptocurrencyService {
-    fn wire_api(&self, builder: &mut ServiceApiBuilder) {
-        CryptocurrencyApi.wire(builder);
+    fn initialize(&self, context: CallContext<'_>, _params: Vec<u8>) -> Result<(), ExecutionError> {
+        Schema::initialize(context.service_data());
```

Comment:
> even writing the tests won't help you there

Why do you think this is the case? For example, it's enough to comment out this line for *all* `cryptocurrency-advanced` tests to fail. And they all fail with "Cryptocurrency schema is not initialized" message, too!

Also, I'm sorry, but "It's pretty easy to forget to initialize the schema" reads like "It's pretty easy to forget to create your tables in RDBMS". I'd say (a) it's not that easy given the right mindset (indexes = RDMBS tables) and (b) the consequences are immediate and obvious. You can no longer materialize an index from thin air, so when the only option API gives you is `Option<Index>`, you should think why this `Option` should or should not be `Some(_)`.

Regarding automated initialization: it could be possible to implement a more typed version of `Service` trait for Rust services, which would know about schema (e.g., have it as an associated type), so that the schema could be initialized automatically. This is outside the scope of this PR.

Reply:
Well, okay, I was wrong about the tests, but as I told: it's just a first glance and not the actual review. To understand that commenting this line will break the tests, you have to compile the Exonum in your head, since this fact relies on the surrounding context.

Your analogy is good, but I'd compare it with raw pointers in the C++: of course you have to init it within constructor, and of course consequences are immediate and obvious if you'll forget to do it. But somehow programs still do fail because of the null pointer dereference. And Rust somehow allows users not to worry about it.

tl;dr: I really love keeping things as simple and stupid as possible, and IMO it's too low-level for a public framework interface. I would really like the situation when users won't have to think why API gives you an `Option`, so they'll be able to focus on the business logic.